lib/bolt/puppetdb/client.rb

```diff
@@ -50,7 +50,8 @@ def make_query(query, path = nil)
 
         begin
           response = http_client.post(url, body: body, header: headers)
-        rescue SocketError, OpenSSL::SSL::SSLError, SystemCallError, Net::ProtocolError, IOError => err
+        rescue SocketError, OpenSSL::SSL::SSLError, SystemCallError,
```

Comment:
maybe we should just rescue StandardError for this? this is the kind of bug I was concerned about when I added this

I'm ok merging if someone thinks thats a bad idea.

Reply:
Updated to raise ` Bolt::PuppetDBFailoverError` for *any* standard error with http response as discussed.